lib/interceptor/dns.js

```diff
@@ -227,11 +254,13 @@ class DNSInstance {
   setRecords (origin, addresses) {
     const timestamp = Date.now()
     const records = { records: { 4: null, 6: null } }
+    let minTTL = this.#maxTTL
     for (const record of addresses) {
       record.timestamp = timestamp
       if (typeof record.ttl === 'number') {
         // The record TTL is expected to be in ms
         record.ttl = Math.min(record.ttl, this.#maxTTL)
+        minTTL = Math.min(minTTL, record.ttl)
```

Comment:
For LRU or any other caches, which will not be able to respect ttl per record, just pass the minimum value from records.

Can't find better solution because `records` is a mutable object and we need to store the same for every origin,

Reply:
ah ok see it now, missed that bit